In [1]:
# 패키지 불러오기

# api 요청
import requests
import json
# 데이터 자료형 및 분석도구
import pandas as pd
import numpy as np
# 시각화 패키지
import matplotlib as mlp
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.image as mpimg
import matplotlib.pyplot as plt
from matplotlib import patheffects

%matplotlib inline
sns.set()
# 스케일링
from sklearn.preprocessing import MinMaxScaler
# 시간
import time
from datetime import datetime
# 진행 사항 확인
from tqdm import tqdm

# PUBG 분석 도구
from chicken_dinner.pubgapi import PUBG
from chicken_dinner.constants import COLORS
from chicken_dinner.constants import map_dimensions
from chicken_dinner.models import tournaments

In [2]:
# api key 설정 및 데이터 요청
api_key = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJqdGkiOiI2ZGNjNzNhMC0yMDk5LTAxMzctYjNjMi0wMmI4NjZkNzliOGIiLCJpc3MiOiJnYW1lbG9ja2VyIiwiaWF0IjoxNTUxNjk2NzA2LCJwdWIiOiJibHVlaG9sZSIsInRpdGxlIjoicHViZyIsImFwcCI6InB1YmctYmVzdC1wbGF5In0.Oz7GmLsWF7038XIO4vKd5sivhLreOnizxTNcARAEFQs'

headers = {'accept': 'application/vnd.api+json',
           'Authorization': f'Bearer {api_key}'}
'''
accountId로 필터 ( 아이디를 알면 ['data']['id'] 를 사용해 accountId 확인가능 ) -> 게임 아이디를 바꿀수있으니 이를 방지할때 accountId사용 
url = "https://api.pubg.com/shards/steam/players/account.573238034f5b4d368c993c5f4d36a2f1" 
r = requests.get(url, headers=headers)
Hwet = r.json()
Hwet['data']['id'] 로 accountId 확인가능
'''

# 아이디로 필터
id = 'Hwet_J'
url = f"https://api.pubg.com/shards/steam/players?filter[playerNames]={id}"
r = requests.get(url, headers=headers)
Hwet = r.json()

# match_list 저장
match_list = [Hwet['data'][0]['relationships']['matches']['data'][count]['id'] for count in range(len(Hwet['data'][0]['relationships']['matches']['data']))]
# print(match_list)


In [5]:
print(match_list)
print(len(match_list))

['b8e3cbb2-6e8b-4b55-bc89-345bbf6c1bbd', '3fb27ecf-868c-424f-be8c-ee64d97fef2a', '4ba6e2f2-c504-4b02-bd2a-b93f3b0b9862', '7169bc8b-9158-4543-be3d-33426826d217', '5b31c032-c484-44ac-9b31-c2c16663924e', '40b3212b-7be5-4bf9-984c-fa2dd2b7ac16', 'c556094b-36ff-4394-a8d4-c59d2b197cc2', '1357fc62-ca25-4210-9cc7-3390f4a96b59', '825d64d8-2d28-4cdf-9b4f-fde7d3d05b5c', '015455da-c331-46fe-9314-9c1ba2d57a36', '690a5909-54d4-4e95-bf88-075eb63f54a4', '9c3b0876-5909-4591-a86d-2df797251ecc', 'b922e78e-61a9-4dcb-8b8d-99bfc50ca369', 'ba3fba65-0ca7-49f5-975e-6922dbabf6d3', 'c23cf640-75ed-4e1b-bde4-47be704f5874', 'd9a18138-fbca-40a6-945e-841ccc53449e', 'a4801fa3-6920-423b-ad25-b82e9ea6b511', '80c80c1c-39bb-48c8-89ef-84ecf837e123', '25c4dbc4-b5b9-42c8-b37e-3cbd29f75109', 'd4a75c8a-6afe-4595-99c7-9830d4d69a13', '2f897f47-37d6-4ff9-b653-6a3298c8f690', '14b0b9e3-93a5-4012-8790-13908dab28ca', '1ef566b9-4008-4a48-9440-7119f3ceb6eb', 'eb52c5d0-5835-46bc-9ff8-47eeeb7d5804', '90fbc033-739f-4bd9-86d6-3e571822c243',

In [20]:
# apply lambda를 위한 함수 (시리즈 변환)
def ParticipantsMerge(row):
    participants = pd.Series(row['attributes']['stats'])
    participants['id'] = row['id']
    return participants

mat_id = match_list[0]
url = f'https://api.pubg.com/shards/steam/matches/{mat_id}'
r = requests.get(url, headers=headers)
data = r.json()
print(data['data'])
print()
# print(data['data']['attributes']['matchType'])  # competitive
# print(data['data']['attributes']['createdAt'])  # 2022-03-24T14:24:21Z
print()
included = pd.DataFrame(data['included'])
rosters = included[included['type'] == 'roster']
rosters = rosters.apply(lambda row: [row['attributes']['stats']['rank'],
                                     row['attributes']['stats']['teamId'],
                                     row['attributes']['won'],
                                     pd.DataFrame(row['relationships']['participants']['data']),
                                     row['id']], axis=1)
rosters = pd.DataFrame(list(rosters), columns=['rank', 'teamId', 'won', 'participants', 'id'])
# print(rosters)
# print(data['data']['relationships']['rosters'])

# match_dict = {match['attributes']['createdAt']: mat_id for match in data['data'] if 'competitive' in match['matchType']}
participants = included[included['type'] == 'participant']
participants = participants.apply(lambda row: ParticipantsMerge(row), axis=1)
participants = participants.reset_index(drop=True)
print(participants['playerId'])
participants = participants.drop(['killPoints', 'killPointsDelta', 'lastKillPoints', 'lastWinPoints',
                                  'mostDamage', 'rankPoints', 'winPoints', 'winPointsDelta'],
                                  axis=1, errors='ignore')


{'type': 'match', 'id': 'b8e3cbb2-6e8b-4b55-bc89-345bbf6c1bbd', 'attributes': {'shardId': 'steam', 'tags': None, 'mapName': 'Tiger_Main', 'seasonState': 'progress', 'duration': 1655, 'gameMode': 'squad', 'titleId': 'bluehole-pubg', 'matchType': 'competitive', 'createdAt': '2022-03-24T14:24:21Z', 'stats': None, 'isCustomMatch': False}, 'relationships': {'rosters': {'data': [{'type': 'roster', 'id': 'c3086f76-978d-4055-83eb-9d0b1030ce38'}, {'type': 'roster', 'id': '3d1aa4c7-3115-4ff1-b423-0fae24fc085c'}, {'type': 'roster', 'id': 'ea9330fc-63b2-4944-bb20-ba20a4159bed'}, {'type': 'roster', 'id': '467bb45e-13b0-473f-a7be-ab717671ff01'}, {'type': 'roster', 'id': '57fa82e6-0241-4ed7-ba7b-313f8f31fcad'}, {'type': 'roster', 'id': 'ac5c2004-9d58-4725-86c6-cb2820baaeab'}, {'type': 'roster', 'id': '8906bb50-1447-4673-b3cb-983e87648869'}, {'type': 'roster', 'id': '48190326-c101-4716-8e4a-e96509501d1f'}, {'type': 'roster', 'id': '0c567b2d-b175-4684-a14c-0ffb638c7b97'}, {'type': 'roster', 'id': 'd581

Writing default config to: C:\Users\HC\.jupyter\jupyter_notebook_config.py


<p>print(data['data']['attributes'])</p>
{'shardId': 'steam', 'tags': None, 'mapName': 'Tiger_Main', 'seasonState': 'progress', 'duration': 1655, 'gameMode': 'squad', 'titleId': 'bluehole-pubg', 'matchType': 'competitive', 'createdAt': '2022-03-24T14:24:21Z', 'stats': None, 'isCustomMatch': False}